### PR TITLE
Feature branch with bubble_sort methods 

### PR DESCRIPTION
**Pull request by RossielCS**
This branch counts with bubble_sort and bubble_sort_by methods.
We used the bubble sort algorithm and for that, we applied while loops and booleans.

### CODE REVIEW

File: main.rb [hunk 1]
```diff
@@ -0,0 +1,36 @@
...
+        arr[i], arr[i + 1] = arr[i + 1], arr[i]
+        i += 1
+        swapp = true
+      elsif arr[i].length <= arr[i + 1].length
+        i += 1
+      end
+    end
+    i = 0
+  end
+  yield arr
```
Comment (maelfosso) [inline main.rb]: Make it returning the sorted array instead

**Review by maelfosso — CHANGES_REQUESTED** [review]
# Changes required :red_circle: 

Hello @RossielCS @Maclenn77 
Nice job done!!!
To move on, please, make `bubble_sort_by` returns the sorted array.
Once done, please, submit a new code review request.

Thanks
Mael FOSSO.

File: main.rb [hunk 2]
```diff
@@ -0,0 +1,34 @@
+def bubble_sort(numbers)
+  switch = true
+  n = numbers.length - 1
+  while switch == true
```
Comment (shshamim63) [inline main.rb]: well, the switch contains a boolean value. So you don't need to compare it with ==
```ruby
while switch  #can do the trick.
```

File: main.rb [hunk 3]
```diff
@@ -0,0 +1,34 @@
...
+      if yield(arr[i], arr[i + 1]).positive? == true
+        arr[i], arr[i + 1] = arr[i + 1], arr[i]
+        swapp = true
+      end
+      i += 1
+    end
+    i = 0
+  end
+  arr
+end
```
Comment (shshamim63) [inline main.rb]: While I was trying to catch some errors, I just refactored the bubble_sort_by a little. It does not reflect too much change though. But you can take a look at it.
```ruby
def bubble_sort_by(arr)
  i = 0
  swapp = true
  while swapp
    swapp = false
    while !arr[i + 1].nil?
      if yield(arr[i], arr[i + 1]).positive?
        arr[i], arr[i + 1] = arr[i + 1], arr[i]
        swapp = true
      end
      i += 1
    end
    i = 0
  end
  arr
end
```

**Review by shshamim63 — APPROVED** [review]
## Status :100: Approved :tada: 
___
Hi @RossielCS and @Maclenn77 :wave: ,
awesome job :clap: . You have covered all the changes asked by the previous reviewer. The repository contains a descriptive readme. So nothing is blocking me right now to approve it. Just got some suggestion to make. In ruby, if we are dealing with variables or build-in methods that are/return boolean, we can use them directly in condition/terminal condition for the loop. Check my comments below. And they are optional changes.
___
#### This :+1: is from me. You can go ahead.
___
## :herb: Happy Coding :herb: 
### Shakhawat Hossain
[Linkedin](https://www.linkedin.com/in/shakhawathossainshamim/)

**Pull request merged**